data-node/src/main/java/org/graylog/datanode/configuration/TlsConfigurationSupplier.java

```diff
@@ -29,41 +28,42 @@ public class TlsConfigurationSupplier {
+        return getTlsConfig(configSubPart, certificateMetaData);
     }
 
-    public void addHttpTlsConfig(Map<String, String> config,
-                                 final CertificateMetaData certificateMetaData) {
+    public ImmutableMap<String, String> getHttpTlsConfig(final CertificateMetaData certificateMetaData) {
         final String configSubPart = SSL_PREFIX + "http";
-        addTlsConfig(config, configSubPart, certificateMetaData);
-        config.put(configSubPart + ".enabled", "true");
+        return ImmutableMap.<String, String>builder()
+                .putAll(getTlsConfig(configSubPart, certificateMetaData))
+                .put(configSubPart + ".enabled", "true")
+                .build();
     }
 
-    public void addTrustStoreTlsConfig(Map<String, String> config,
-                                       final String truststorePassword) {
+    public ImmutableMap<String, String> getTrustStoreTlsConfig(final String truststorePassword) {
+        final ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+
         String configSubPart = SSL_PREFIX + "http";
-        config.put(configSubPart + ".truststore_type", TRUSTSTORE_FORMAT);
-        config.put(configSubPart + ".truststore_filepath", TRUSTSTORE_FILENAME);
-        config.put(configSubPart + ".truststore_password", truststorePassword);
+        builder.put(configSubPart + ".truststore_type", TRUSTSTORE_FORMAT);
+        builder.put(configSubPart + ".truststore_filepath", TRUSTSTORE_FILENAME);
+        builder.put(configSubPart + ".truststore_password", truststorePassword);
 
         configSubPart = SSL_PREFIX + "transport";
-        config.put(configSubPart + ".truststore_type", TRUSTSTORE_FORMAT);
-        config.put(configSubPart + ".truststore_filepath", TRUSTSTORE_FILENAME);
-        config.put(configSubPart + ".truststore_password", truststorePassword);
+        builder.put(configSubPart + ".truststore_type", TRUSTSTORE_FORMAT);
+        builder.put(configSubPart + ".truststore_filepath", TRUSTSTORE_FILENAME);
+        builder.put(configSubPart + ".truststore_password", truststorePassword);
+        return builder.build();
     }
 
-    private void addTlsConfig(Map<String, String> config,
-                              final String configPrefix,
-                              final CertificateMetaData certificateMetaData
+    private ImmutableMap<String, String> getTlsConfig(final String configPrefix,
+                                                      final CertificateMetaData certificateMetaData
     ) {
-        config.put(configPrefix + ".keystore_type", KEYSTORE_FORMAT);
-        config.put(configPrefix + ".keystore_filepath", certificateMetaData.keystoreFilePath());
-        config.put(configPrefix + ".keystore_password", certificateMetaData.passwordAsString());
-        config.put(configPrefix + ".keystore_alias", certificateMetaData.alias());
+        return ImmutableMap.<String, String>builder()
```

Comment:
Aren't generics in this line redundant?

Reply:
I thought as well, but it doesn't compile without. So we need them here.